# 연습문제2 풀이

## 패키지 참조

In [1]:
from scipy import stats
from pandas import read_excel

## 문제 1

### 데이터 가져오기

In [2]:
df = read_excel("https://data.hossam.kr/E02/material.xlsx")
df

상표1   상표2
0   80.4  80.1
1   78.2  81.2
2   80.1  79.5
3   77.1  78.0
4   79.6  76.1
5   80.4  77.0
6   81.6  80.1
7   79.9  79.9
8   84.4  78.8
9   80.9  80.8
10  83.1   NaN

### 가설 설정

| 가설 | 내용 | 식 |
|--|--|--|
| 귀무가설 | 상표1과 상표2의 주성분 함량에는 차이가 없다. | $H_0 : \mu_1 = \mu_2$ |
| 대립가설 | 상표1과 상표2의 주성분 함량에는 차이가 있다. | $H_0 : \mu_1 \neq \mu_2$ |


### 분석1

In [5]:
df.dropna(inplace=True)
stats.ttest_ind(df['상표1'],df['상표2'], equal_var=True, alternative='two-sided')

TtestResult(statistic=1.3710671216714043, pvalue=0.18720759661580608, df=18.0)

### 결과

검정통계량 `1.371`에서 `p-value`가 유의수준 `0.05`보다 크므로 귀무가설을 기각할 수 없다. 즉, 상표1과 상표2의 주성분 함량에는 차이가 없다.

### 분석2

In [9]:
df1 = df.fillna(df.mean())
df1

상표1   상표2
0  80.4  80.1
1  78.2  81.2
2  80.1  79.5
3  77.1  78.0
4  79.6  76.1
5  80.4  77.0
6  81.6  80.1
7  79.9  79.9
8  84.4  78.8
9  80.9  80.8

In [11]:
# equal_var=True 생략 가능
stats.ttest_ind(df['상표1'],df['상표2'], equal_var=True, alternative='two-sided')

TtestResult(statistic=1.3710671216714043, pvalue=0.18720759661580608, df=18.0)

### 결과

검정통계량(t통계량) `1.371`에서 `p-value(유의확률)`가 `0.187`로 유의수준 `0.05`보다 크므로 귀무가설을 기각할 수 없다. 즉, 상표1과 상표2의 주성분 함량에는 차이가 없다.

## 문제 2

### 데이터 가져오기

In [12]:
df = read_excel("https://data.hossam.kr/E02/blood_pressure.xlsx")
df

복용전  복용후
0    70   68
1    80   72
2    72   62
3    76   70
4    76   58
5    76   66
6    72   68
7    78   52
8    82   64
9    64   72
10   74   74
11   92   60
12   74   74
13   68   72
14   84   74

### 가설 설정

| 가설 | 내용 | 식 |
|--|--|--|
| 귀무가설 | 피임약 복용전과 복용후 혈압은 차이가 없다.<br/>피임약은 혈압에 영향을 주지 않는다. | $H_0 : \mu_1 = \mu_2$ |
| 대립가설 | 피임약 복용전과 복용후 혈압은 차이가 있다.<br/>피임약은 혈압에 영향을 준다. | $H_0 : \mu_1 \neq \mu_2$ |

### 분석

In [16]:
# equal_var=True 불가
stats.ttest_rel(df['복용전'], df['복용후'], alternative='two-sided')

TtestResult(statistic=3.105360487466109, pvalue=0.007749436106654119, df=14)

### 결과

검정통계량(t통계량) `3.105`에서 `p-value(유의확률)`가 `0.008`로 유의수준 `0.05`보다 작기 때문에 귀무가설을 기각하고 대립가설을 채택한다. 즉, 피임약은 혈압에 영향을 준다고 볼 수 있다.

## 문제 3

### 데이터 가져오기

In [17]:
df = read_excel("https://data.hossam.kr/E02/runner_diet.xlsx")
df

감량전   감량후    
0  14.5  14.3 NaN
1  13.4  13.4 NaN
2  13.0  13.1 NaN
3  14.5  13.9 NaN
4  13.3  13.1 NaN
5  14.2  13.9 NaN
6  13.6  13.6 NaN
7  13.9  14.0 NaN
8  13.2  12.8 NaN
9  13.0  12.8 NaN

In [18]:
df.dropna(axis=1, inplace=True)
df

감량전   감량후
0  14.5  14.3
1  13.4  13.4
2  13.0  13.1
3  14.5  13.9
4  13.3  13.1
5  14.2  13.9
6  13.6  13.6
7  13.9  14.0
8  13.2  12.8
9  13.0  12.8

### 가설 설정

| 가설 | 내용 | 식 |
|--|--|--|
| 귀무가설 | 체중 감량은 달리기 결과에 영향이 없다. | $H_0 : \mu_1 = \mu_2$ |
| 대립가설 | 체중 감량은 달리기 결과에 영향이 있다. | $H_0 : \mu_1 \neq \mu_2$ |

In [19]:
stats.ttest_rel(df['감량전'], df['감량후'], alternative='two-sided')

TtestResult(statistic=2.375306807446375, pvalue=0.04154604923393679, df=9)

### 결과

검정통계량(t통계량) `2.375`에서 `p-value(유의확률)`가 `0.042`로 유의수준 `0.05`보다 작기 때문에 귀무가설을 기각하고 대립가설을 채택한다. 즉, 체중 감량은 달리기 속도에 영향을 준다고 볼 수 있다.

## 문제 4

### 데이터 가져오기

In [20]:
df = read_excel("https://data.hossam.kr/E02/wrestler_diet.xlsx")
df

이름  감량전 악력  감량후 악력  감량전 윗몸일으키기  감량후 윗몸일으키기  감량전 턱걸이  감량후 턱걸이
0   B.S      43      41          35          41       25       29
1   P.J      42      41          40          44       25       29
2   G.G      52      50          36          41       22       23
3   A.L      53      52          38          42       10       15
4   I.H      44      40          36          42       15       19
5   B.Y      46      47          39          44       17       19
6   I.K      48      47          35          42       24       26
7   K.B      58      55          34          36       17       18
8   K.J      50      50          33          35       17       20
9   S.S      48      50          39          45       21       25
10  S.P      56      48          37          37       31       35
11  S.J      43      41          33          37       15       22

### 결과값에서 `e-05` 주의

In [21]:
stats.ttest_rel(df['감량전 악력'], df['감량후 악력'], alternative='two-sided')

TtestResult(statistic=2.3654081485744824, pvalue=0.037454446305019644, df=11)

In [22]:
stats.ttest_rel(df['감량전 윗몸일으키기'], df['감량후 윗몸일으키기'], alternative='two-sided')

TtestResult(statistic=-7.179926973956724, pvalue=1.798236117030195e-05, df=11)

In [23]:
stats.ttest_rel(df['감량전 턱걸이'], df['감량후 턱걸이'], alternative='two-sided')

TtestResult(statistic=-6.841977654731714, pvalue=2.7937021187235204e-05, df=11)

### 반복문을 적용하여 풀어보기

In [24]:
df2 = df.set_index('이름')
cols = df2.columns
cols

Index(['감량전 악력', '감량후 악력', '감량전 윗몸일으키기', '감량후 윗몸일으키기', '감량전 턱걸이', '감량후 턱걸이'], dtype='object')

In [25]:
# 변수 길이 확인
n = len(cols)
n

6

In [28]:
# 2단위로 컬럼을 사용
for i in range(0, n, 2):
    # print(i, i+1)
    result = stats.ttest_rel(df2[cols[i]], df2[cols[i+1]], alternative='two-sided')
    print('\n', result)


 TtestResult(statistic=2.3654081485744824, pvalue=0.037454446305019644, df=11)

 TtestResult(statistic=-7.179926973956724, pvalue=1.798236117030195e-05, df=11)

 TtestResult(statistic=-6.841977654731714, pvalue=2.7937021187235204e-05, df=11)
